source/agora/network/Manager.d

```diff
@@ -570,22 +571,27 @@ public class NetworkManager
                     const ckey = key;
                     try
                     {
-                        auto payload = this.registry.getValidatorInternal(ckey);
-                        if (payload == RegistryPayloadData.init)
-                        {
-                            log.warn("Could not find mapping in registry for key {}", ckey);
-                            return;
-                        }
-
-                        if (payload.public_key != ckey)
-                        {
-                            log.error("Registry answered with the wrong key: {} => {}",
-                                      ckey, payload);
-                            return;
-                        }
+                        this.registry.getValidatorInternal(ckey,
+                            (in Message msg) @trusted {
```

Comment:
I feel like the `NetworkManager` should not be dealing with DNS `Message`s directly, I still think using `DNSResolver` here (or in `NetworkClient`) is the sanest way forward.

Reply:
I agree that manager shouldn't be dealing with any DNS stuff

Using `DNSResolver` within Manager would end up in following chain (for caching node)
```
Manager -> DNSResolver -> Registry (checks cache) -> DNSResolver -> out network reach other node
```
and it would just be, not so necessary, additional step. We can still resolve dealing with DNS stuff in manager directly using the provided registry interface but it would end up with array allocations. Instead of requiring a delegate it can return array of addresses directly.